openassessment/xblock/openassessmentblock.py

```diff
@@ -336,11 +336,22 @@ def file_upload_type(self, value):
         """
         self.file_upload_type_raw = value
 
+    def get_allowed_file_types_or_preset(self):
+        if self.white_listed_file_types:
+            return self.white_listed_file_types
+        elif self.file_upload_type == 'image':
+            return self.ALLOWED_IMAGE_EXTENSIONS
+        elif self.file_upload_type == 'pdf-and-image':
+            return self.ALLOWED_FILE_EXTENSIONS
+        return None
+
     @property
     def white_listed_file_types_string(self):
         """
-        Join the white listed file types into comma delimited string
+        Join the allowed file types into comma delimited string
         """
+        self.white_listed_file_types = self.get_allowed_file_types_or_preset()
```

Comment:
So, if I follow, this is doing the following:
 - If the block has a defined whitelist, use it
 - If the block /doesn't/ have a defined whitelist, grab the defaults and save them on the block so it now does have a whitelist.

I think it would be clearer if we put this logic somewhere line in `StudioMixin.update_editor_context()`. So in the future the whitelist is populated on save.
We can keep this to "backpopulate", but let's add some comments explicitly stating that that's what it's doing.

Reply:
Agreed.

Reply:
^ Updating w/ `update_editor_context` will happen as a part of [EDUCATOR-5344](https://openedx.atlassian.net/browse/EDUCATOR-5344), where we'll fill the allow list when a preset is selected, updating the XBlock on save. My goal here is to have an in-situ way of loading presets for non-updated ORAs.

Reply:
I'm gonna push back, I think this is the ticket for this change, but I'm down to have a discussion in standup tomorrow.